Assets/Scripts/WeakPoint.cs

```diff
@@ -24,6 +25,7 @@ public void Hit(float damageAmount)
         {
             enemyAI.SetPlayerDamageTaken(damageAmount);
             gameObject.SetActive(false);
+            popped = true;
```

Comment:
Instead of setting a flag and making the object inactive, destroy it and tell the AI to remove it from the list of weakpoints